packages/abi-gen/src/index.ts

```diff
@@ -126,18 +125,21 @@ for (const abiFileName of abiFileNames) {
     }
 
     const methodAbis = ABI.filter((abi: AbiDefinition) => abi.type === ABI_TYPE_METHOD) as MethodAbi[];
-    const methodsData = _.map(methodAbis, methodAbi => {
-        _.map(methodAbi.inputs, (input, i: number) => {
+    const methodAbisSanitized = abiUtils.renameOverloadedMethods(methodAbis) as MethodAbi[];
+    const methodsData = _.map(methodAbis, (methodAbi, i: number) => {
+        _.map(methodAbi.inputs, (input, j: number) => {
```

Comment:
conceptually `map` should be used for side-effect free conversion from an array of one type into another type. if you want to iterate  over a data source and mutate it you should use `forEach` instead